### PR TITLE
feat(ui-ux): add header updates to OCG

### PR DESCRIPTION
**Pull request by veralygit**
#### What this PR does / why we need it:
This PR updates the header for scan and adds OCG to header

#### Which issue(s) does this PR fixes?:

Fixes #

#### Sample Links & Screenshots:

Link:

<details>
<summary>Desktop Screenshot</summary>

</details>
<details>
<summary>Mobile Screenshot</summary>

</details>

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] Tested on multiple web browsers
- [ ] Tested responsiveness (e.g, iPhone, iPad, Desktop)
- [ ] No console errors
- [ ] Unit tests\*
- [ ] Added e2e tests\*

### CODE REVIEW

File: src/layouts/components/Header.tsx [hunk 1]
```diff
@@ -142,71 +205,96 @@ function DesktopNavbar(): JSX.Element {
...
 
-      <div className="p-2 bg-primary-700 dark:bg-gray-900 flex flex-wrap p-4 md:p-0">
-        <HeaderCountBar className="w-full flex flex-wrap" />
-        <div className="w-full mt-4">
+function MobileMenu({ toggleMenu }: { toggleMenu: () => void }): JSX.Element {
+  const ref = useRef<HTMLDivElement>(null);
+  const dimension = useWindowDimensions();
+  useEffect(() => {
+    if (ref.current) {
+      ref.current.style.height = `${dimension.height - 225}px`;
```
Comment (kyleleow) [inline src/layouts/components/Header.tsx:257]: how do you get `225`?

Comment (veralygit) [reply src/layouts/components/Header.tsx:257]: updated to use ref.current.offsetTop to calculate the height

File: src/layouts/components/Header.tsx [hunk 2]
```diff
@@ -34,103 +39,167 @@ export function Header(): JSX.Element {
...
         />
+
+        {/* <HeaderLink
+          className="ml-1 lg:ml-4"
+          text="Consortium"
+          pathname="/consortium/asset_breakdown"
+          testId="Desktop.HeaderLink.Consortium"
+        /> */}
+
+        {useWindowDimensions().width <= 1530 ? (
```
Comment (fullstackninja864) [inline src/layouts/components/Header.tsx:193]: can we declare width variable outside JSX element?  hooks should get used outside JSX element

File: src/layouts/components/Header.tsx [hunk 3]
```diff
@@ -216,17 +312,24 @@ function HeaderLink(props: {
...
+            "text-gray-900 dark:text-dark-gray-900":
+              router.pathname !== props.pathname,
+          }
+        )}
         data-testid={props.testId}
       >
         <div
           className={classNames(
-            "inline m-2 text-lg pb-0.5 hover:text-primary-500 dark:hover:text-dark-50  cursor-pointer",
+            "dark:hover:text-dark-50 m-2 inline cursor-pointer pb-0.5 text-lg  hover:text-primary-500",
```
Comment (fullstackninja864) [inline src/layouts/components/Header.tsx:330]:
```suggestion
            "dark:hover:text-dark-50 m-2 inline cursor-pointer pb-0.5 text-lg hover:text-primary-500",
```

File: src/layouts/components/Header.tsx [hunk 4]
```diff
@@ -237,3 +340,145 @@ function HeaderLink(props: {
...
+              })}
+            </Menu.Items>
+          </Transition>
+        </>
+      )}
+    </Menu>
+  );
+}
+
+const dropDownLinks = [
```
Comment (fullstackninja864) [inline src/layouts/components/Header.tsx:473]: In future if we want to add new menu item the we have to update the entry at two places
1. at [link](https://github.com/WavesHQ/scan/blob/1a5994a882e99f84bf69687556eeeabf45b10dbf/src/layouts/components/Header.tsx#L196)
2. in dropDownLinks array [link](https://github.com/WavesHQ/scan/blob/1a5994a882e99f84bf69687556eeeabf45b10dbf/src/layouts/components/Header.tsx#L473)
I guess we can check a diff logic to address this

**Review by fullstackninja864 — CHANGES_REQUESTED** [review]

File: src/layouts/components/Header.tsx (hunk 2 above)
Comment (veralygit) [reply src/layouts/components/Header.tsx]: removed check for width tab will be grouped under `More`

File: src/layouts/components/Header.tsx (hunk 3 above)
Comment (veralygit) [reply src/layouts/components/Header.tsx]: updated

File: src/layouts/components/Header.tsx (hunk 4 above)
Comment (veralygit) [reply src/layouts/components/Header.tsx:456]: For desktop tabs, new tabs will now be grouped under `More` dropdown there wouldnt be 2 points to update. Just at dropDownLinks array checks for width of screen is removed

File: src/layouts/components/Header.tsx [hunk 5]
```diff
@@ -237,3 +323,145 @@ function HeaderLink(props: {
     </Link>
   );
 }
+
+function MenuItems({ viewPort }: { viewPort: string }): JSX.Element {
+  return (
+    <div className="flex flex-col">
+      <HeaderLink
+        className="flex justify-start border-b border-[#E5E5E5] dark:border-gray-700 md:pt-0 px-4 p-1.5"
```
Comment (kyleleow) [inline src/layouts/components/Header.tsx:331]:
```suggestion
        className="flex justify-start border-b border-gray-200 dark:border-gray-700 md:pt-0 px-4 p-1.5"
```

File: src/layouts/components/Header.tsx [hunk 6]
```diff
@@ -34,103 +39,150 @@ export function Header(): JSX.Element {
...
-    >
-      <div className="hidden md:block border-b border-gray-100 bg-primary-700 dark:bg-gray-800 dark:border-0">
-        <Container className="py-1">
-          <div className="flex items-center justify-between h-8">
-            <HeaderCountBar className="h-full flex" />
-            <HeaderNetworkMenu />
-          </div>
-        </Container>
-      </div>
+  if (menu) {
```
Comment (kyleleow) [inline src/layouts/components/Header.tsx:42]: can we wrap this in `useEffect`?

File: src/layouts/components/Header.tsx [hunk 7]
```diff
@@ -237,3 +323,145 @@ function HeaderLink(props: {
...
+            enterTo="transform scale-100 opacity-100"
+            leave="transition duration-75 ease-out"
+            leaveFrom="transform scale-100 opacity-100"
+            leaveTo="transform scale-95 opacity-0"
+          >
+            <Menu.Items className="absolute m-4 min-w-max flex flex-col divide-y bg-white border rounded-lg border-gray-200 dark:border-gray-700 dark:bg-gray-700">
+              {dropDownLinks.map((item, index) => {
+                return (
+                  <Menu.Item key={index}>
+                    <>
```
Comment (kyleleow) [inline src/layouts/components/Header.tsx:430]: we need this?

File: src/layouts/components/Header.tsx [hunk 8]
```diff
@@ -237,3 +323,145 @@ function HeaderLink(props: {
...
+                      <Link href={{ pathname: item.link }} passHref>
```
Comment (kyleleow) [inline src/layouts/components/Header.tsx:431]: can you check how to close the `More` dropdown after user click on `On-Chain Governance`?

File: src/layouts/components/Header.tsx [hunk 9]
```diff
@@ -34,103 +39,152 @@ export function Header(): JSX.Element {
...
-          </div>
-        </Container>
-      </div>
-
-      <>{menu && <MobileMenu />}</>
-    </header>
+          </Container>
+        </div>
+      </header>
+      <>
```
Comment (kyleleow) [inline src/layouts/components/Header.tsx:125]: do we need this?

File: src/layouts/components/Header.tsx [hunk 10]
```diff
@@ -237,3 +325,143 @@ function HeaderLink(props: {
...
+  //   rootPathName: "consortium",
+  // },
+  {
+    name: "On-Chain Governance",
+    link: "/on-chain-governance",
+    rootPathName: "on-chain-governance",
+  },
+];
+
+const menuItemLinks = [
```
Comment (kyleleow) [inline src/layouts/components/Header.tsx:430]: can we rename to `drawerMenuItemLinks`? to indicate that this is only for non-desktop

File: src/layouts/components/Header.tsx (hunk 9 above)
Comment (veralygit) [reply src/layouts/components/Header.tsx]: removed

File: src/layouts/components/Header.tsx (hunk 10 above)
Comment (veralygit) [reply src/layouts/components/Header.tsx]: renamed

**Pull request merged**